packages/grid/_modules_/grid/models/index.ts

```diff
@@ -1,6 +1,5 @@
 export * from './colDef';
 export * from './gridContainerProps';
-export * from './elementSize';
```

Comment:
Not sure of the benefit of removing that.
If ppl want to have the type of `state.containerProps.viewportSize` then they would need to recreate it 🤔

Reply:
How about we hid this state too before the beta release? This way, we force developers to report the problem they face, and we can, on a case by case basis make the state public or propose an ever better solution? This would allows us to refactor the internal of the grid, before the next breaking change window.

Thinking more out loud here. I even wonder if the core and x shouldn't have different breaking change timing policy. Right now, it's every 12 months for the core. Maybe 6 months for x would be better.

Reply:
I think this is a separate issue. Could we open it separately. For this change, I would be in favour of rolling back as it's a different issue, and if we take another approach we would have an element missing in our index.

We have a different vision here, in brief, exposing the state is all about showing the internal state of the grid to advanced users or dev tools...

Reply:
Agree that it's a different issue and that we could bring value by simply namespacing. We can handle how to optimize for learning on the data grid, and maximize for refactoring flexibility in a different issue 👍 . In [my initial comment](https://github.com/mui-org/material-ui-x/pull/1292#issuecomment-808110770), I was mentioning "at some point" 😁.

Reply:
> We have a different vision here, in brief, exposing the state is all about showing the internal state of the grid to advanced users or dev tools...

Which I advocate is harmful to the product. But, I also think that trying it out and coming to this conclusion on our own is valuable for the person leading the effort. I think that it's will be an expensive lesson to learn but an important one. And who knows. Maybe I'm wrong, and it will work out just fine. 

It seems to be a generic product owner/manager strategy question. I personally think that getting the product to be as light as possible, focusing on the good enough, and forcing developers to open an issue to complain about what's missing is the most effective and efficient approach.

Reply:
Alright, I'll revert the change. We can tackle it in a separate issue that is broader and includes hiding other things from the public API.